### PR TITLE
Feature/auth0

### PR DESCRIPTION
**Pull request by berthj193**
I want to merge this change because...

### Pull Request Checklist

1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] Changes are mentioned in the changelog.

### Test Environment Config

API_URI=https://qa.staging.saleor.cloud/graphql/

### CODE REVIEW

File: src/views/Home/View.tsx [hunk 1]
```diff
@@ -22,6 +24,16 @@ const View: React.FC<NextPage> = () => {
    * be added in SALEOR-1566.
    */
   React.useEffect(() => {
+    const getToken = async () => {
+      const claims = await getIdTokenClaims();
+      const idToken = claims && claims["__raw"];
```
Comment (marianmazarovici) [inline src/views/Home/View.tsx]: what's this?

Comment (berthj193) [reply src/views/Home/View.tsx]: I tried to output `id_token`, but I think `getAccessTokenSilently` will be enough.

https://github.com/auth0/auth0-react#call-an-api

**Comment by marianmazarovici** [comment]
@mingthegit let me know when it's ready for review

**Comment by marianmazarovici** [comment]
Is this ready yet?!

**Comment by berthj193** [comment]
This is ready, dependent on https://github.com/vonq/saleor-storefront/pull/15.

For failing tests, I found that `develop` branch has tests failing too. Investigating why PR#15 introduced this Github check.

**Comment by E1379** [comment]
@mingthegit Do we really noeed to include the sdk this way? Can we not add it to packages: https://github.com/mirumee/saleor-storefront/blob/master/package.json#L11

**Comment by berthj193** [comment]
> Do we really noeed to include the sdk this way? Can we not add it to packages:

It was in package.json, but I wanted a surgery on `@saler/sdk` to customize authentication. That's the reason I started the synchronization and cloned sdk in the repo.

**Comment by E1379** [comment]
> > Do we really noeed to include the sdk this way? Can we not add it to packages:
> 
> It was in package.json, but I wanted a surgery on `@saler/sdk` to customize authentication. That's the reason I started the synchronization and cloned sdk in the repo.

Can we not fork the sdk and add needed changes? If we do that we have at least a way of tracking changes. Maybe we can adjust the SDK to support both "standard" and Ath0 and make PR to original repo? @marianmazarovici What is your take on this?

**Comment by berthj193** [comment]
> Can we not fork the sdk and add needed changes? If we do that we have at least a way of tracking changes. Maybe we can adjust the SDK to support both "standard" and Ath0 and make PR to original repo? @marianmazarovici What is your take on this?

Definitely we can fork the sdk repo, but `@saleor/sdk` is only used in storefront, and I foresee there will be lots of divergence again in that repository since it's in active development. Saleor dashboard has so-called its own `@saleor/sdk` version in the repository itself.

The principle I had in mind was that, the more dependencies, the more difficult to maintain a software.

As Emil and me talked briefly on Slack, we need a detailed discussion on the future of storefront or new JMP.

**Pull request closed without merging**